src/dmd/parse.d

```diff
@@ -8919,15 +8919,32 @@ final class Parser(AST) : Lexer
         endtok = token.value == TOK.leftBracket ? TOK.rightBracket : TOK.rightParentheses;
 
         nextToken();
+
+        const errorsBefore = errorCount;
+        bool newErrors = false;
         while (token.value != endtok && token.value != TOK.endOfFile)
         {
             auto arg = parseAssignExp();
             arguments.push(arg);
             if (token.value == endtok)
                 break;
-            check(TOK.comma);
+            if (token.value != TOK.comma)
```

Comment:
It would be simpler to just break out of the loop for comma, too, and replace the error message for `check(endtok)` below.